benches/common.sh

```diff
@@ -0,0 +1,109 @@
+# will not work if not under root
+# if benches are run localy by a developer droping caches might not be too important
+function maybe_drop_cache {
+	if test -w /proc/sys/vm/drop_caches; then
+		echo 3 > /proc/sys/vm/drop_caches
+	fi
+}
+
+function can_be_run_under_numa {
+	numactl "$@" true 1>/dev/null 2>/dev/null
+}
+
+# not every local machine has numa enabled on it
+# not every local machine has enough hardware units to run with intended configuration
+#
+# usage:
+# maybe_under_numactl numaoptions -- command to run
+# example:
+# maybe_under_numactl --membind=1 --cpunodebind=1 --physcpubind=11 -- echo qwe
+function maybe_under_numactl {
+	local numaoptions=()
+	local cmdoptions=()
+	local parsing_numa=1
+	for option in "$@"; do
+		if [ -n "$parsing_numa" ]; then
+			if [ "$option" == -- ]; then
+				parsing_numa=
+			else
+				numaoptions+=( "$option" )
+			fi
+		else
+			cmdoptions+=( "$option" )
+		fi
+	done
+
+	if which numactl 1>/dev/null 2>/dev/null; then
+		# check if it is even runable with given numactl options
+		if can_be_run_under_numa "${numaoptions[@]}"; then
+			numactl "${numaoptions[@]}" "${cmdoptions[@]}"
+		else
+			"${cmdoptions[@]}"
+		fi
+	else
+		"${cmdoptions[@]}"
+	fi
+}
+
+function get_tarantool_version {
+	"$TARANTOOL_EXECUTABLE" -v | grep -e "Tarantool" |  grep -oP '\s\K\S*'
+}
+
+function clean_tarantool {
+	rm -rf ./*.snap
+	rm -rf ./*.xlog
+	rm -rf ./*.vylog
+	rm -rf 5*
+}
+
+function sync_disk {
+	sync && echo "sync passed" || echo "sync failed with error: $?"
+}
+
+function kill_tarantool {
+	killall tarantool 2>/dev/null || true
+}
+
+function error {
+	echo "ERROR:" "$@"
+	exit 100
+}
+
+function stop_and_clean_tarantool {
+	kill_tarantool
+	clean_tarantool
+	sync_disk
+}
+
+function wait_for_file_release {
+	local f="$1"
+	local t="$2"
+	local tt=0
+	while [ "$tt" -lt "$t" ]; do
+		if ! lsof "$f" 1>/dev/null 2>/dev/null; then
+			return 0
+		fi
+
+		tt=$(( tt + 1 ))
+		sleep 1
+	done
+
+	return 1
+}
+
+function wait_for_tarantool_runnning {
+	local creds="$1"
+	local t="$2"
+	local tt=0
+
+	while [ "$tt" -lt "$t" ]; do
+		if echo 'if type(box.cfg) ~= "function" then return box.info().status end' | tarantoolctl connect "$creds" 2>/dev/null | grep -q 'running'; then
```

Comment:
I don't ready to say whether we can lean on `type(box.cfg)` and `box.info().status` here. I suggest to use your own variable / file / whatever to wait a point, where all tarantool-side initialization is done.

Reply:
I would like to use this approach in the future when working on fixing all the benchmarks mess.

Reply:
What blocks us to do it right now?

Reply:
Mostly not wanting to go through a lot of patching with current gitlab-ci pipeline in mind (that should not be broken). However this could be easily done in separate branches with all install.sh scripts targeting this specific branch. If patching is required to be done in such a manner that it must be merged in master and not break the current pipeline, I am afraid this will only make the code even worse.

Reply:
I would really like to use separate branches as it could greatly improve code quality and remove a lot of ugly setup code from run.sh scripts.